src/code/db_camera.c

```diff
@@ -1,284 +1,1453 @@
+    dbCamera->sub.unk_104A = dbCamera->sub.unk_104E;
     dbCamera->fov = 0.0f;
     dbCamera->unk_50 = 0.0f;
-    D_80161100 = camera->globalCtx;
-    dbCamera->unk_7C = 0;
-    dbCamera->unk_7E = -1;
-    dbCamera->unk_80 = 1;
-    dbCamera->unk_82 = 0;
-    dbCamera->unk_84 = 0;
-    dbCamera->unk_86 = 0;
-    dbCamera->unk_78 = 0;
-    dbCamera->unk_7A = 0;
-    dbCamera->unk_10C0.x = 0;
-    dbCamera->unk_10C0.y = 0;
-    dbCamera->unk_10C0.z = 0;
-    dbCamera->unk_6C = dbCamera->unk_70 = dbCamera->unk_74 = 0;
-}
-
-void DbgCamera_Enable(DBCamera* dbCamera, Camera* camera) {
-    dbCamera->at = camera->at;
-    dbCamera->eye = camera->eye;
-    dbCamera->up = camera->up;
-    dbCamera->fov = camera->fov;
-    dbCamera->unk_4C = 0;
-    dbCamera->unk_80 = 1;
-    dbCamera->unk_82 = 0;
-    dbCamera->unk_84 = 0;
-    dbCamera->unk_86 = 1;
-    dbCamera->unk_88 = 1;
+    sGlobalCtx = cameraPtr->globalCtx;
+    dbCamera->sub.mode = 0;
+    dbCamera->sub.nFrames = -1;
+    dbCamera->sub.nPoints = 1;
+    dbCamera->sub.unkIdx = 0;
+    dbCamera->sub.unk_08 = 0;
+    dbCamera->sub.unk_0A = 0;
     dbCamera->unk_78 = 0;
     dbCamera->unk_7A = 0;
-    dbCamera->unk_50 = 0.0f;
-    func_800B4088(dbCamera, camera);
+    dbCamera->sub.demoCtrlMenu = DEMO_CTRL_MENU(ACTION_E, MENU_INFO);
+    dbCamera->sub.demoCtrlActionIdx = ACTION_E;
+    dbCamera->sub.demoCtrlToggleSwitch = 0;
+    dbCamera->unk_6C.x = 0;
+    dbCamera->unk_6C.y = 0;
+    dbCamera->unk_6C.z = 0;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/db_camera/DbgCamera_Update.s")
+s32 DbgCamera_Enable(DbCamera* this, Camera* cam) {
+    this->at = cam->at;
+    this->eye = cam->eye;
+    this->unk_1C = cam->up;
+    this->fov = cam->fov;
+    this->roll = 0;
+    this->sub.nPoints = 1;
+    this->sub.unkIdx = 0;
+    this->sub.unk_08 = 0;
+    this->sub.unk_0A = 1;
+    this->sub.unk_0C = 1;
+    this->unk_78 = 0;
+    this->unk_7A = 0;
+    this->unk_50 = 0.0f;
+    return func_800B4088(this, cam);
+}
 
-s32 func_800B8730(void) {
-    s32 temp_v1;
-    void* phi_v0;
-    s32 phi_v1;
+#pragma GLOBAL_ASM("asm/non_matchings/code/db_camera/DbCamera_Update.s")
 
-    for (phi_v1 = 0; phi_v1 < 0x1A; phi_v1++) {
-        if (D_801612D0[phi_v1] != 'O') {
-            return phi_v1 + 'A';
+s32 DbCamera_GetFirstAvailableLetter() {
+    s32 i;
+    for (i = 0; i < ARRAY_COUNT(sLetters); i++) {
+        switch (sLetters[i]) {
+            case 'O':
+                break;
+            default:
+                return 'A' + i;
         }
     }
+
     return '?';
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/db_camera/func_800B87D8.s")
+char DbCamera_InitCut(s32 idx, DbCameraSub* sub) {
+    s32 i;
 
-/**
- * Clears a DBCSEditorSlot
- */
-void func_800B8978(s32 editorSlot, s32 free) {
+    sDbCameraCuts[idx].unk_01 = 0x61;
+    sDbCameraCuts[idx].letter = DbCamera_GetFirstAvailableLetter();
+    D_80161250[0x3F + sDbCameraCuts[idx].letter] = 'O';
```

Comment:
This seems really odd for a few reasons. 

1. I can't find where `D_80161250` is referenced in meaningful code 
2. DbCamera_GetFirstAvailableLetter's minimum return value is 0x3F, and it's expected return value is minimum 0x41. 0x3F + 0x41 = 0x80. D_80161250 + 0x80 = D_801612D0, or sLetters.